### DISCUSSION TITLE
How can I give different random outputs "weight" so one happens more often than the others?

### DISCUSSION THREAD

**Discussion by Sklartacus** [inkle/ink#813] [Writing in ink] [answered]
Hello! 

So, I know how to create a coin flip. I could write a shuffle, say, 
{~Heads|Tails} 
and it will produce either "Heads" or "Tails." 

However, what if I had a coin that was weighted, such that it landed on "Heads" 90% of the time? I could brute force it, like 
{~Heads|Heads|Heads|Heads|Heads|Heads|Heads|Heads|Heads|Tails}
but that feels really inelegant, and would frustrate what I want to use the result of that coin flip for. 

So - is there a way to add weight to one side of the coin? Or to have certain random events occur with frequency ("The town is quiet today.") and others occur very rarely ("The town is overrun with dinosaurs!")?

As I write this out, I think I might try creating random number tables instead of plain shuffles. As in, 
~ temp dice_roll = RANDOM(1, 10)
and then have numbers 1-9 trigger one thing, and 10 trigger the other. Is that the best way to go about this?

**Comment by joningold** [comment, marked as answer]
Think of shuffles as a short-hand: a quick way to solve a common problem. If you want to make your solution more specific, then break out a bit more scripting. Storing a dice-roll and testing it is a great way to achieve what you want, and get all the control over the weights you need.

**Reply by Sklartacus** [reply]
Thanks for replying (and for everything else)!
